### PR TITLE
Fix slicing to actually drop and recreate WCS as intended

### PR DESCRIPTION
**Pull request by rosteen**
From the comments in the code, it looks like the intent here was always to create a new lookup table SpectralGWCS when slicing, but the `copy` method defaults to `self.wcs` and thus sliced spectra were ending up with the original spectrum's WCS. This doesn't solve the fact that spatial WCS components may end up incorrect if the spatial dimensions are sliced, but it at least fixes the spectral part. We may be able to fall back more on the NDCube machinery for spatial slicing than we are now, I'll open a follow-up issue for that. I'll also add a test for this case shortly.

### CODE REVIEW

**Comment by rosteen** [comment]
See https://github.com/astropy/specutils/issues/919 for followup to do with spatial WCS axes.

**Comment by keflavich** [comment]
lgtm but I'm not giving formal approval b/c of my persistent unfamiliarity w/the code base.  Agree, though, copies should have copies of WCS!

**Review by eteq — APPROVED** [review]
I think part of the thinking here is that the original `wcs` has some context information that should get copied over. That said, I totally agree this is an improvement, because the current behavior is way more confusing.

I wonder, though, if the previous wcs should remain, say, in `meta` with a name like `'previous_wcs'` or something? I could be argued either way, though, as I can see good or bad sides.  Happy to merge this once you've decided either way on this, @rosteen .

**Comment by rosteen** [comment]
@eteq I decided to take your suggestion and retain it in `meta` as `original_wcs`, mainly for the case where someone might override their original spectrum with the slice and thus lose access to the original WCS.

**Comment by eteq** [comment]
Sounds good. I have a minor niggle that this should be documented more clearly somewhere, but... since this wasn't documented *before* I'm thinking to merge this and make that better in #919.

**Pull request merged**
